File: src/data/courses/dsa/merge-sort/sorting-comparisons/internal-vs-external/code/python/internal_vs_external.py
# Internal Sorting
def internal_sort(arr):
    arr.sort()  # All data in memory

# External Sorting
def external_sort(file, chunk_size):
    # 1. Divide into chunks
    chunks = divide_into_chunks(file, chunk_size)
    # 2. Sort each chunk in memory
    for chunk in chunks:
        sort_chunk(chunk)
    # 3. Merge sorted chunks
    merge_chunks(chunks)

File: src/data/courses/dsa/merge-sort/sorting-comparisons/internal-vs-external/code/java/InternalVsExternal.java
// Internal Sorting (in-memory)
void internalSort(int[] arr) {
    Arrays.sort(arr); // All data in memory
}

// External Sorting (disk-based)
class ExternalSort {
    void externalSort(String file, int chunkSize) {
        // 1. Divide into chunks
        List<String> chunks = divideIntoChunks(file, chunkSize);
        // 2. Sort each chunk in memory
        for (String chunk : chunks) sortChunk(chunk);
        // 3. Merge sorted chunks
        mergeChunks(chunks);
    }
}